canvas/src/main/scala/doodle/canvas/algebra/Immediate.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2015 Creative Scala
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package doodle.canvas.algebra
+
+import doodle.core.Color
+import org.scalajs.dom.CanvasRenderingContext2D
+
+trait Immediate {
+  def fillColor(color: Color)(implicit ctx: CanvasRenderingContext2D): Unit = {
+    ctx.fillStyle = CanvasDrawing.colorToCSS(color)
+  }
+
+  def rectangle(x: Double, y: Double, width: Double, height: Double)(implicit ctx: CanvasRenderingContext2D): Unit = {
+    ctx.fillRect(x, y, width, height)
```

Comment:
This does not match the standard Doodle coordinates, where we use the *center* of the shapes, not the top-left corner.